app/components/animated_number/index.tsx

```diff
@@ -26,43 +26,43 @@ const AnimatedNumber = ({
     animateToNumber,
     animationDuration,
     fontStyle,
-    easing,
+    easing = 1.2,
 }: Props) => {
     const prevNumber = usePrevious(animateToNumber);
     const animateToNumberString = String(Math.abs(animateToNumber));
     const prevNumberString = String(Math.abs(prevNumber));
 
-    const numberStringToDigitsArray = Array.from(animateToNumberString, Number);
-    const prevNumberersArr = Array.from(prevNumberString, Number);
-
     const [numberHeight, setNumberHeight] = React.useState(0);
-    const animations = useMemo(() => numberStringToDigitsArray.map((__, index) => {
-        if (typeof prevNumberersArr[index] !== 'number') {
-            return new Animated.Value(0);
+    const animations = useMemo(() => {
+        if (numberHeight === 0) {
+            return [];
         }
+        const numberStringToDigitsArray = Array.from(animateToNumberString, Number);
+        const prevNumberersArr = Array.from(prevNumberString, Number);
 
-        const animationHeight = -1 * (numberHeight * prevNumberersArr[index]);
-        return new Animated.Value(animationHeight);
-    }), [numberStringToDigitsArray]);
+        return numberStringToDigitsArray.map((digit, index) => {
+            // Skip animation if the current and previous digits are the same
+            if (prevNumberersArr[index] === digit) {
+                return new Animated.Value(-1 * (numberHeight * digit));
```

Comment:
Yes, this condition could indeed cause an issue when transitioning from 99 to 100. The problem arises because we skip the animation when the digits are the same. For example:

- `prevNumberersArr = [9, 9]`
- `numberStringToDigitsArray = [1, 0, 0]`

When comparing the first digit (`1` vs `9`), the condition `prevNumberersArr[0] === digit` will fail, which is expected, and the animation will proceed for the first digit. However, for the second and third digits, the condition `prevNumberersArr[1] === digit` will compare `9` (previous) to `0` (current), which is not equal, so those digits will still be animated, which is correct.

The issue is not necessarily in this block itself, but the logic could behave unexpectedly if we are rapidly changing values (e.g., every 100ms). Skipping digits could lead to an incorrect visual transition, especially when the change is rapid, such as going from `99` to `100`.

To ensure the conditions for rapid changes are avoided we can add an extra condition for checking whether the length of the `prevNumberersArr` and `numberStringToDigitsArray` are equal or not which will ensure that the animation isn't skipped due to rapid changes.